adapters/insticator/insticator.go

```diff
@@ -0,0 +1,335 @@
+package insticator
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"strings"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+)
+
+type ext struct {
+	Insticator impInsticatorExt `json:"insticator"`
+}
+
+type impInsticatorExt struct {
+	AdUnitId    string `json:"adUnitId"`
+	PublisherId string `json:"publisherId"`
+}
+
+type adapter struct {
+	endpoint string
+}
+
+type reqExt struct {
+	Insticator *reqInsticatorExt `json:"insticator,omitempty"`
+}
+
+type reqInsticatorExt struct {
+	Caller []insticatorCaller `json:"caller,omitempty"`
+}
+
+type insticatorCaller struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+}
+
+// caller Info used to track Prebid Server
+// as one of the hops in the request to exchange
+var caller = insticatorCaller{"Prebid-Server", "n/a"}
+
+type bidExt struct {
+	Insticator bidInsticatorExt `json:"insticator,omitempty"`
+}
+
+type bidInsticatorExt struct {
+	MediaType string `json:"mediaType,omitempty"`
+}
+
+// Builder builds a new instance of the Insticator adapter with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+// getMediaTypeForBid figures out which media type this bid is for
+func getMediaTypeForBid(bid *openrtb2.Bid) openrtb_ext.BidType {
+	switch bid.MType {
+	case openrtb2.MarkupBanner:
+		return openrtb_ext.BidTypeBanner
+	case openrtb2.MarkupVideo:
+		return openrtb_ext.BidTypeVideo
+	default:
+		return openrtb_ext.BidTypeBanner
+	}
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var adapterRequests []*adapters.RequestData
+	var groupedImps = make(map[string][]openrtb2.Imp)
+
+	reqExt, err := makeReqExt(request)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	request.Ext = reqExt
+
+	// Create a deep copy of the request to avoid modifying the original request
+	requestCopy := *request
+	requestCopy.Site = request.Site
+	requestCopy.App = request.App
```

Comment:
This code is making a shallow copy of the request, not a deep copy as mentioned in the comment. This is fine as it protects the original request from changes in `populatePublisherId`. Since Site and App are pointers, those two lines are no-ops and can be removed.